proxy/src/transport/tls/config.rs

```diff
@@ -250,6 +318,24 @@ impl ServerConfig {
     }
 }
 
+pub fn current_connection_config<C>(identity: Option<&Identity>, watch: &Watch<Option<C>>)
+    -> ConditionalConnectionConfig<C> where C: Clone + std::fmt::Debug
+{
+    match identity {
```

Comment:
TIOLI: I find the multiple nested matches somewhat unclear. 

I think this could also be expressed with `Option::and_then` or matching on `(identity, *watch.borrow())`, though your approach does have the advantage of not borrowing the watch when there's no identity...

Reply:
If we were using `Result` instead of `Conditional`, this could also be represented as

```rust
identity.and_then(|identity| {
    *watch.borrow()
        .and_then(|ref config| Some(ConnectionConfig {
            identity: identity.clone(),
            config: config.clone()
        })
        .ok_or(ReasonForNoTls::NoConfig)
}).ok_or(ReasonForTls::Disabled)
```

Reply:
`and_then` ...`ok_or` isn't clearer than the nested matches.

Note that this is a generalization of the nested `if let Some(...)` construct we already have in src/connection.rs in `BoundPort::listen_and_fold`. I wrote this to be generic in the hope that we'd be able to clean up `BoundPort::listen_and_fold` using this.